ksql-engine/src/main/java/io/confluent/ksql/schema/registry/SchemaRegistryUtil.java

```diff
@@ -64,7 +68,24 @@ public static void cleanUpInternalTopicAvroSchemas(
   public static void deleteSubjectWithRetries(
       final SchemaRegistryClient schemaRegistryClient,
       final String subject) throws Exception {
-    ExecutorUtil.executeWithRetries(() -> schemaRegistryClient.deleteSubject(subject), ALWAYS);
+    ExecutorUtil.executeWithRetries(() -> {
+      try {
+        schemaRegistryClient.deleteSubject(subject);
```

Comment:
Kind of a small point, but we could make this endpoint snappier if we just return immediately if we get a 403 or 401 (or any 4xx for that matter). Retries certainly won't help in those cases. We could either have the function return an Optional<Exception> and throw if we get something back. Or change `executeWithRetries` to support a whitelist of exceptions to not retry on.